### PR TITLE
fix(http.js): add config to disable http responses capture

### PR DESCRIPTION
**Pull request by ranrib**
temp workaround for https://github.com/epsagon/epsagon-node/issues/256

### CODE REVIEW

File: src/events/http.js [hunk 1]
```diff
@@ -374,8 +378,10 @@ function httpWrapper(wrappedFunction) {
 
                 clientRequest.on('response', (res) => {
                     // Listening to data only if options.epsagonSkipResponseData!=true or no options
-                    if (!options || (options && !options.epsagonSkipResponseData)) {
-                        res.on('data', chunk => addChunk(chunk, chunks));
+                    if (!config.getConfig().disableHttpResponseBodyCapture) {
+                        if (!options || (options && !options.epsagonSkipResponseData)) {
+                            res.on('data', chunk => addChunk(chunk, chunks));
+                        }
```
Comment (galbash) [inline src/events/http.js]: use && ?

Comment (ranrib) [reply src/events/http.js]: 👍

File: src/events/http.js [hunk 2]
```diff
@@ -279,7 +279,11 @@ function httpWrapper(wrappedFunction) {
                 this, buildParams(url, options, patchedCallback)
             );
 
-            if (options && options.epsagonSkipResponseData) {
+            if (
+                options &&
+                options.epsagonSkipResponseData &&
```
Comment (tripodsan) [inline src/events/http.js:284]: shouldn't that be:

`if (options && options.epsagonSkipResponseData || config.getConfig().disableHttpResponseBodyCapture)` ?

Comment (ranrib) [reply src/events/http.js:284]: We want to enable this tracing for fetch-h2 only if all is true:
1. `options` exists.
2. `options.epsagonSkipResponseData` is set to true
3. Nobody disabled the response capture: `!config.getConfig().disableHttpResponseBodyCapture`

This handling is specific to fetch-h2. Later in the module, there's an handling for the rest of the libs.

**Review by galbash — APPROVED** [review]

File: src/events/http.js (hunk 2 above)
Comment (tripodsan) [reply src/events/http.js:284]: but this is exactly what isn't working. the shimmer doesn't detect fetch-h2, and doesn't set the `options.epsagonSkipResponseData`. so we can force it with `disableHttpResponseBodyCapture`..maybe I don't understand the flags....

Comment (tripodsan) [reply src/events/http.js:284]: anyways...it seems to work now :-)

**Pull request merged**